### DISCUSSION TITLE
Editing a YAML document through a non-text editor

### DISCUSSION THREAD

**Discussion by FALLAI-Denis** [redhat-developer/vscode-yaml#664] [Q&A]
Hi,

We would like to develop an extension with a specialized (non-text) editor to handle a YAML document.
The YAML document would not be directly visible in the editor (graphic editor or HTML page type), on the other hand the YAML document will have to be validated through a Schema, and the errors will have to go up to the editor.

Do you know of any extensions that work on this principle?
Or advice for the development of our extension?

Thank you.

**Comment by gorkem** [comment]
I think [vscode-tekton](https://github.com/redhat-developer/vscode-tekton) has a graphical pipeline builder that may fit the bill.  @evidolob can tell more.

**Reply by evidolob** [reply]
Not a builder, just viewer.
We don't have graphical editing for yaml in vscode-tekton.
